nautobot/core/signals.py

```diff
@@ -64,5 +64,11 @@ def invalidate_max_depth_cache(sender, **kwargs):
     """Clear the appropriate TreeManager.max_depth cache as the create/update/delete may have changed the tree."""
     from nautobot.core.models.tree_queries import TreeManager
 
-    if isinstance(sender.objects, TreeManager) and hasattr(sender.objects, "max_depth"):
+    if not isinstance(sender.objects, TreeManager):
+        return
+
+    # Note that we can't use `hasattr`, because this will trigger `max_depth` to be calculated if it doesn't exist
+    try:
```

Comment:
It looks like `__dict__.pop` is actually more performant here and more concise. Thoughts?

```py
# test cached_property invalidation speed
import functools
import timeit


class TestCache:
    @functools.cached_property
    def max_depth(self):
        return 1000


def invalidate_cache_del():
    t = TestCache()
    t.max_depth
    del t.max_depth


def invalidate_cache_pop():
    t = TestCache()
    t.max_depth
    t.__dict__.pop("max_depth", None)


def invalidate_cache_novalue_del():
    t = TestCache()
    try:
        del t.max_depth
    except AttributeError:
        pass


def invalidate_cache_novalue_pop():
    t = TestCache()
    t.__dict__.pop("max_depth", None)


def main():
    print("del:", timeit.timeit(invalidate_cache_del, number=1000000))
    print("pop:", timeit.timeit(invalidate_cache_pop, number=1000000))
    print("del no value:", timeit.timeit(invalidate_cache_novalue_del, number=1000000))
    print("pop no value:", timeit.timeit(invalidate_cache_novalue_pop, number=1000000))


if __name__ == "__main__":
    main()
```

Output:
```
del: 0.5046068249794189
pop: 0.571504608989926
del no value: 0.423761794983875
pop no value: 0.14525465900078416
```

Reply:
LGTM

Reply:
Looks small enough to be not a big factor either way, but I'm fine with either.